libs/data-access/api/src/lib/trpc/routers/settings.router.ts

```diff
@@ -18,6 +18,22 @@ export const settingsRouter = t.router({
 			})
 		)
 		.mutation(async ({ ctx, input }) => {
+			const currentSettings = await database.studentSettings.findUnique({
+				where: {
+					username: ctx.user.name
+				}
+			});
+
+			if (currentSettings?.hasLearningDiary !== input.settings.hasLearningDiary) {
+				await database.eventLog.create({
+					data: {
+						type: "LTB_TOGGLE",
+						payload: { enabled: input.settings.hasLearningDiary },
+						username: ctx.user.name
```

Comment:
etwas kontrovers, allerdings würde ich hier, die userId oder username auf die ressourceId setzen. 

`username` ist der "invoker" (evtl wird auch mal der name geändert). Die resourceId ist die geänderte ressource. 

Wenn wir später protokollieren, falls ein admin für einen anderen nutzer einstellungen ändert, wird das so auch erfasst. 

sind alles edge cases, allerdings mehr consistent

Reply:
Wenn das in einem passiert, wird aber auch ein Toggle Event aufgerufen, wenn man Statistiken aktiviert. Dann wird der aktuelle Status des LTB (ausgeschaltet) noch einmal als Event in den Log geschrieben. Wenn das kein Problem ist, kann ich das in eins tun. Sonst nicht.